Extensions/permalink_copy.js

```diff
@@ -0,0 +1,124 @@
+//* TITLE permalink_copy **//
+//* VERSION 1.0.0 **//
+//* DESCRIPTION **//
+//* DEVELOPER New-XKit **//
+//* FRAME false **//
+//* BETA false **//
+
+
+let lastTarget;
+
+const copyArea = document.createElement("input");
+copyArea.type = "text";
+copyArea.style = "position: absolute; top: -99em; left:0;";
+
+/*
+ * only run for links of type permalink. LMB: Hook this way to ensure browser compatibility. 
+ */
+function shouldRun(target) {
+	return (target && target.nodeName === "A" && target.classList.contains('permalink'));
+}
+
+function onPermalinkClick(event) {
+
+	if (!shouldRun(event.target)) return;
+
+	const swallowNewWindow = "auxclick" !== event.type;
+	const prefs = XKit.extensions.permalink_copy.preferences;
+	let preference = swallowNewWindow ? prefs.activate : prefs.activate_mmb;
+	if (!!preference && !preference.value) {
+		return;
+	}
+
+	// prevent button from opening a new tab
+	preference = prefs.no_prevent_default;
+
+	if (swallowNewWindow && !(preference && preference.value)) {
+		event.preventDefault();
+	}
+
+	lastTarget = event.target;
+
+	document.body.appendChild(copyArea);
+	const link = lastTarget.href;
+	copyArea.value = link;
+	copyArea.select();
+	const success = document.execCommand('Copy');
+	
+	if (XKit.extensions.permalink_copy.preferences.show_toast.value) {
+		XKit.notifications.add(success ? "permalink copied to clipboard" : "couldn't copy link");
+	}	
+	document.body.removeChild(copyArea);
+	return false;
+}
+
+
+
+function interceptClickEvent(event) {	 
+	if (shouldRun(event.target)) onPermalinkClick(event);
+}
+
+function interceptMmbClickEvent(event) { 
+	if (shouldRun(event.target)) onPermalinkClick(event);
+}
+
+XKit.extensions.permalink_copy = new Object({
```

Comment:
Please move the preference definition to the top of the file (under the header) for consistency

Reply:
I moved the object definition, I hope that was what you meant. The linter wouldn't let me move the preferences outside of the class.